### PR TITLE
Fixed: Xml signature validation

### PR DESCRIPTION
**Pull request by gekiss**
- Fixed: XML signature validation error if signature element has namespace prefix.
- Fixed: XML signature validation error if signature contain `InclusiveNamespaces` element.
- Implemented a test that validates the above fixed issue.
#1003

### CODE REVIEW

**Comment by gekiss** [comment]
@dotnet-policy-service agree

File: src/CoreWCF.Primitives/src/CoreWCF/Security/XmlHelper.cs [hunk 1]
```diff
@@ -411,5 +411,40 @@ public static UniqueId ReadTextElementAsUniqueId(XmlElement element)
...
             return new UniqueId(ReadTextElementAsTrimmedString(element));
         }
+
+        internal static string[] TokenizeInclusiveNamespacesPrefixList(string inclusiveNamespacesPrefixList)
+        {
+            if (inclusiveNamespacesPrefixList == null)
+                return null;
+
+            string[] prefixes = inclusiveNamespacesPrefixList.Split(null);
+            int count = 0;
```
Comment (mconnew) [inline src/CoreWCF.Primitives/src/CoreWCF/Security/XmlHelper.cs:421]: You can simplify this code by using the Split overload which takes StringSplitOptions and pass RemoveEmptyEntries. This way the returned array will always be the correct length and you only need to replace "#default" with String.Empty. That allows you to drop the prefix.Length check and the resize code at the end.

File: src/CoreWCF.Primitives/src/CoreWCF/Security/WSSecurityOneDotZeroReceiveSecurityHeader.cs [hunk 2]
```diff
@@ -125,8 +125,21 @@ protected override SignedXml ReadSignatureCore(XmlDictionaryReader signatureRead
                 writer.WriteEndDocument();
             }
             SignedXMLInternal signedXml = new SignedXMLInternal(doc);
-            XmlNodeList nodeList = doc.GetElementsByTagName("Signature");
+            XmlNodeList nodeList = doc.GetElementsByTagName("Signature", SignedXml.XmlDsigNamespaceUrl);
             signedXml.LoadXml((XmlElement)nodeList[0]);
+            if (signedXml.SignedInfo.CanonicalizationMethodObject is XmlDsigExcC14NTransform xmlDsigExcC14NTransform)
+            {
```
Comment (mconnew) [inline src/CoreWCF.Primitives/src/CoreWCF/Security/WSSecurityOneDotZeroReceiveSecurityHeader.cs:131]: I tried to do some validation looking at the behavior with our existing tests and it looks like we usually have XmlDsigC14NTransform as our transform so any problems here won't be picked up by our existing tests. The main thing I'm concerned with here is that signatureReader might not have a namespace matching the prefix when you do the lookup. Can you null check the value returned from LookupNamespace and only add if it isn't null.

**Comment by mconnew** [comment]
Just a couple of small changes and we can accept this PR.

File: src/CoreWCF.Primitives/src/CoreWCF/Security/XmlHelper.cs [hunk 3]
```diff
@@ -411,5 +411,27 @@ public static UniqueId ReadTextElementAsUniqueId(XmlElement element)
         {
             return new UniqueId(ReadTextElementAsTrimmedString(element));
         }
+
+        internal static string[] TokenizeInclusiveNamespacesPrefixList(string inclusiveNamespacesPrefixList)
+        {
+            if (inclusiveNamespacesPrefixList == null)
+                return null;
+
+            string[] prefixes = inclusiveNamespacesPrefixList.Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
```
Comment (mconnew) [inline src/CoreWCF.Primitives/src/CoreWCF/Security/XmlHelper.cs:420]: You can pass `null` as the first parameter. The "The separator array" section of the documentation says:

> If the separator parameter is null or contains no characters, white-space characters are assumed to be the delimiters.

You could also pass `Array.Empty<char>` if there's problems with ambiguous overloads.
```suggestion
            string[] prefixes = inclusiveNamespacesPrefixList.Split(null, StringSplitOptions.RemoveEmptyEntries);
```

**Comment by mconnew** [comment]
I should be releasing this next week if no blockers come up.

**Pull request merged**